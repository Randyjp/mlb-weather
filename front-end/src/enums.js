import {Enum} from 'enumify';

export class UnitSystem extends Enum {
}

UnitSystem.initEnum(['METRIC', 'IMPERIAL']);


export const TemperatureScales = Object.freeze({
    METRIC: "Celsius",
    IMPERIAL: "Fahrenheit",
});

export const SpeedScales = Object.freeze({
    METRIC: "meters/second",
    IMPERIAL: "miles/hour",
});

