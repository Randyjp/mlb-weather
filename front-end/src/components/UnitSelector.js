import React from 'react';
import PropTypes from 'prop-types';

import {UnitSystem} from '../enums';

import {Button, ButtonGroup, ButtonOr} from 'semantic-ui-react';

const UnitSelector = ({handleUnitChange, units}) => (
    <ButtonGroup basic>
        <Button
            onClick={() => handleUnitChange(UnitSystem.METRIC)}
            active={units === UnitSystem.METRIC}>
            °C
        </Button>
        <ButtonOr/>
        <Button
            onClick={() => handleUnitChange(UnitSystem.IMPERIAL)}
            active={units === UnitSystem.IMPERIAL}>
            °F
        </Button>
    </ButtonGroup>

);

export default UnitSelector;