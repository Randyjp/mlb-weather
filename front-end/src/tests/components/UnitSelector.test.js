import React from 'react';
import {shallow} from 'enzyme';
import {Button} from 'semantic-ui-react';
import renderer from 'react-test-renderer';

import {UnitSystem} from '../../enums';
import UnitSelector from '../../components/UnitSelector';

it('renders without crashing', () => {
    shallow(<UnitSelector units={UnitSystem.METRIC} handleUnitChange={() => {
    }}/>);
});

it('should render correctly', () => {
    const tree = renderer.create(<UnitSelector handleUnitChange={()=>{}}
                                        units={UnitSystem.IMPERIAL}/>);
    expect(tree.toJSON()).toMatchSnapshot();

});

describe('<UnitSelector/> component', () => {
    let wrapper;
    const spy = jest.fn();

    beforeEach(() => {

        wrapper = shallow(<UnitSelector
            units={UnitSystem.METRIC}
            handleUnitChange={spy}
        />);

    });


    it('should render two <Button> components', () => {
        expect(wrapper.find(Button)).toHaveLength(2);
    });

    it('simulates click events', () => {
        wrapper.find(Button).first().simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should have active class on right button ', () => {
        const first = wrapper.find(Button).first();
        const second = wrapper.find(Button).last();

        expect(first.props().active).toBeTruthy();
        expect(second.props().active).toBeFalsy();
    });
});

