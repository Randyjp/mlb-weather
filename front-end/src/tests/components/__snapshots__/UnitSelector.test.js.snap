// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="ui basic buttons"
>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    °C
  </button>
  <div
    className="or"
    data-text={undefined}
  />
  <button
    className="ui active button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    °F
  </button>
</div>
`;
